### PR TITLE
nrf5x/rustdoc

### PR DESCRIPTION
**Pull request by niklasad1**
Hi,
This tries to solve [#605](https://github.com/helena-project/tock/issues/605)

I totally is stuck on ARM SysTick so I figured got to make some progress today.

This PR contribute with following:
* modified the ```//!``` comments to look *ok" after running `rustdoc`.   
* in some files I have made minor changes

Thoughts?

### CODE REVIEW

File: boards/nrf51dk/src/main.rs [hunk 1]
```diff
@@ -3,35 +3,38 @@
...
-//!  14 -> P0.19   (mid right header)
-//!  15 -> P0.18   (mid right header)
-//!  16 -> P0.17   (mid right header)
-//!  17 -> P0.16   (mid right header)
-//!  18 -> P0.15   (mid right header)
-//!  19 -> P0.14   (mid right header)
-//!  20 -> P0.13   (mid right header)
-//!  21 -> P0.12   (mid right header)
+//! ### GPIO pins
+//! --------------
```
Comment (bradjc) [inline boards/nrf51dk/src/main.rs]: Don't need this line.

File: boards/nrf51dk/src/main.rs [hunk 2]
```diff
@@ -3,35 +3,38 @@
...
+//! * 18 -> P0.15   (mid right header)
+//! * 19 -> P0.14   (mid right header)
+//! * 20 -> P0.13   (mid right header)
+//! * 21 -> P0.12   (mid right header)
 //!
-//!  Author: Philip Levis <pal@cs.stanford.edu>
-//!  Author: Anderson Lizardo <anderson.lizardo@gmail.com>
-//!  Date: August 18, 2016
+//! ### Authors
+//! ---------------
```
Comment (bradjc) [inline boards/nrf51dk/src/main.rs]: Remove.

File: boards/nrf51dk/src/main.rs [hunk 3]
```diff
@@ -3,35 +3,38 @@
...
+//! * Philip Levis <pal@cs.stanford.edu>
+//! * Anderson Lizardo <anderson.lizardo@gmail.com>
+//! Date: August 18, 2016
```
Comment (bradjc) [inline boards/nrf51dk/src/main.rs]: Probably want "*" here.

File: chips/nrf51/src/radio.rs [hunk 4]
```diff
@@ -1,14 +1,17 @@
-//! Radio/BLE Driver for nrf51dk
+//! Radio Driver Bluetooth Low Energy nRF51
 //!
 //! Sending BLE advertisement packets
 //! Possible payload is 31 bytes
```
Comment (bradjc) [inline chips/nrf51/src/radio.rs]: Need periods for this paragraph.

File: chips/nrf51/src/radio.rs [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
 //!
 //! Currently all fields in PAYLOAD array are configurable from user-space
 //! except the PDU_TYPE.
 //!
-//! Author: Niklas Adolfsson <niklasadolfsson1@gmail.com>
-//! Author: Fredrik Nilsson <frednils@student.chalmers.se>
-//! Date: June 22, 2017
+//! ### Authors
+//! -----------------
```
Comment (bradjc) [inline chips/nrf51/src/radio.rs]: Remove

File: chips/nrf52/src/radio.rs [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
 //! Sending BLE advertisement packets
 //! Possible payload is 31 bytes
 //!
 //! Currently all fields in PAYLOAD array are configurable from user-space
 //! except the PDU_TYPE.
 //!
-//! Author: Niklas Adolfsson <niklasadolfsson1@gmail.com>
-//! Date: July 18, 2017
+//! ### Author
+//! ------------
```
Comment (bradjc) [inline chips/nrf52/src/radio.rs]: Remove

File: chips/nrf52/src/radio.rs [hunk 7]
```diff
@@ -1,13 +1,15 @@
-//! Radio/BLE Driver for nRF52-DK
+//! Radio driver Bluetooth Low Energy nRF52
 //!
 //! Sending BLE advertisement packets
 //! Possible payload is 31 bytes
```
Comment (bradjc) [inline chips/nrf52/src/radio.rs]: Periods.

File: chips/nrf5x/src/ble_advertising_driver.rs [hunk 8]
```diff
@@ -1,69 +1,73 @@
...
 //!
 //! Only start and send are asynchronous and need to use the busy flag.
 //! However, the synchronous calls such as set tx power, advertisement interval
 //! and set payload can only by performed once the radio is not active.
 //! The reason why is that they can be interleaved by an interrupt
 //!
-//! ---ALLOW SYSTEM CALL ------------------------------------------------------------
-//! Each AD TYP corresponds to an allow number from 0 to 0xFF which is matched
+//! ### Allow system call
+//! ----------------------
```
Comment (bradjc) [inline chips/nrf5x/src/ble_advertising_driver.rs]: Remove.

File: chips/nrf5x/src/ble_advertising_driver.rs [hunk 9]
```diff
@@ -1,69 +1,73 @@
...
-//! ---SUBSCRIBE SYSTEM CALL----------------------------------------------------------
+//! * SUCCESS: The buffer has successfully been filled
+//! * ENOSUPPORT: Invalid allow_num
+//! * ENOMEM: No sufficient memory available
+//! * EINVAL: Invalid address of the buffer or other error
+//! * EBUSY: The driver is currently busy with other tasks
+//! * ENOSUPPORT: The operation is not supported
+//!
+//! ### Subscribe system call
+//! --------------------------
```
Comment (bradjc) [inline chips/nrf5x/src/ble_advertising_driver.rs]: Remove.

File: chips/nrf5x/src/ble_advertising_driver.rs [hunk 10]
```diff
@@ -1,69 +1,73 @@
...
-//!        * ENOMEM:    Not sufficient amount memory
-//!        * EINVAL:    Invalid operation
-//! ------------------------------------------------------------------------------
 //!
-//! ---COMMAND SYSTEM CALL------------------------------------------------------------
+//! * ENOMEM:    Not sufficient amount memory
+//! * EINVAL:    Invalid operation
+//!
+//! ### Command system call
+//! ------------------------
```
Comment (bradjc) [inline chips/nrf5x/src/ble_advertising_driver.rs]: Remove.

File: chips/nrf5x/src/ble_advertising_driver.rs [hunk 11]
```diff
@@ -1,69 +1,73 @@
...
 //!
-//! Author: Niklas Adolfsson <niklasadolfsson1@gmail.com>
-//! Author: Fredrik Nilsson <frednils@student.chalmers.se>
-//! Date: June 22, 2017
+//! * SUCCESS:      The command was successful
+//! * EBUSY:        The driver is currently busy with other tasks
+//! * ENOSUPPORT:   The operation is not supported
+//!
+//! ### Authors
+//! --------------
```
Comment (bradjc) [inline chips/nrf5x/src/ble_advertising_driver.rs]: Remove.

**Comment by brghena** [comment]
This looks good to me at this point. Anything you are still waiting on @bradjc, @niklasad1 or is it good to merge?

**Pull request merged**